### PR TITLE
Notify user that a duplicate tag has been entered

### LINKED ISSUE #198: chrome extension magically removes duplicate tags
When adding tags to a new bookmark via the Chrome extension, and the same tag already exists, it just removes the duplicate tag I just typed with no warning why.

Example typing:
tag1 <space> tag2 <space> tag1 <space> at this point the second tag1 just gets deleted.

It would be nice for some visual indication as to why it just got removed. e.g. blink the first tag with the same name

### PR DESCRIPTION
**Pull request by suryatech**
Fixes #198. 

Fade out the original tag and retain the recently entered tag

### CODE REVIEW

**Comment by mitechie** [comment]
Very very cool. Thanks and will try this out. Maybe I can convince Matt to try it out tonight at CHC.

**Comment by suryatech** [comment]
@mitechie Thanks!

File: bookie/static/js/bookie/tagcontrol.js [hunk 1]
```diff
@@ -204,24 +204,38 @@ YUI.add('bookie-tagcontrol', function (Y) {
...
                         silent: silent
                     });
-
-                // keep this up
-                var unique = true;
-                Y.Array.each(this.get('tags'), function (t) {
-                    if (t.get('text') === current_text) {
-                        unique = false;
+                    
+                // Add necessary YUI modules at the end.
```
Comment (mitechie) [inline bookie/static/js/bookie/tagcontrol.js]: You can remove the comment here. It doesn't add any value at this area of the code.

File: bookie/static/js/bookie/tagcontrol.js [hunk 2]
```diff
@@ -204,24 +204,38 @@ YUI.add('bookie-tagcontrol', function (Y) {
...
+                        return true;
                     }
-                });
+                }, this);
 
-                if (unique) {
-                    this.get('tags').push(new_tag);
-                    // add a new li element before the input one
-                    parent.get('parentNode').insertBefore(new_tag.ui, parent);
+                // Array.each is unncessary. Since we are sure that the tag list
```
Comment (mitechie) [inline bookie/static/js/bookie/tagcontrol.js]: This reads like a great code review comment, but it doesn't help explain the code that's left for future readers. Please remove the comment here. I do agree this is great information to put into the github pull request to aid people reviewing the code in understanding what you've changed and why.

File: bookie/static/js/bookie/tagcontrol.js [hunk 3]
```diff
@@ -204,24 +204,38 @@ YUI.add('bookie-tagcontrol', function (Y) {
...
+                // matches the latest tag would suffice.
+                // Remove the unique flag which is unnecessary. 
 
-                    // fire an event that a new tag was added
-                    if (!silent) {
-                        this.set('events_waiting', true);
-                    }
+                this.get('tags').push(new_tag);
+
+                // add a new li element before the input one
```
Comment (mitechie) [inline bookie/static/js/bookie/tagcontrol.js]: please treat comments as complete sentences that start with a capital letter and end in a period for punctuation. It's something I'm trying to do better with myself.

**Comment by mitechie** [comment]
This looks really good. If you can update the comments and add the test as we talked about in IRC I'd love to get this landed.

**Comment by mitechie** [comment]
Can you rebase this up? The diff is really confused. There's code in there from things that have landed in trunk and it makes it hard to sort through.

**Comment by mitechie** [comment]
superceded

**Pull request closed without merging**